### DISCUSSION TITLE
WSA does not start

### DISCUSSION THREAD

**Discussion by AlexanderSLoburev** [LSPosed/MagiskOnWSALocal#322] [Q&A]
Greetings, I had to reinstall WSA several times because each time it stopped starting at some point. The WSA startup window appears, hangs for a long time, then just closes, and that's the end of it. What could be the cause of this and what steps, other than a complete reinstall, can I take to fix the WSA?

**Comment by Howard20181** [comment]
Please provide more information.

**Reply by AlexanderSLoburev** [reply]
Okay. But I really don't know what information I should provide. I just tried repair and reset (Apps>Installed apps>Windows Subsystem for Android) and also re-run the start.bat script, but that didn't help.

**Reply by Howard20181** [reply]
Create an issue

**Reply by Soetsu09** [reply]
> Greetings, I had to reinstall WSA several times because each time it stopped starting at some point. The WSA startup window appears, hangs for a long time, then just closes, and that's the end of it. What could be the cause of this and what steps, other than a complete reinstall, can I take to fix the WSA?

I faced the same issue as you describe, but it has been solved.
Try to set the GPU used for subsystem to discrete graphics card.
The setting is located in  >>  Windows Subsystem for Android Settings  >>  System >> Graphics and Performance  >>  (Choose your discrete graphics card. Don't select the dynamic mode)

**Reply by AlexanderSLoburev** [reply]
> > Greetings, I had to reinstall WSA several times because each time it stopped starting at some point. The WSA startup window appears, hangs for a long time, then just closes, and that's the end of it. What could be the cause of this and what steps, other than a complete reinstall, can I take to fix the WSA?
> 
> I faced the same issue as you describe, but it has been solved. Try to set the GPU used for subsystem to discrete graphics card. The setting is located in >> Windows Subsystem for Android Settings >> System >> Graphics and Performance >> (Choose your discrete graphics card. Don't select the dynamic mode)

Thank you, friend! Everything is really stable now.

**Reply by Almighty188** [reply]
I have the same issue.  The MagiskOnWSA back in July 2022 where it was built on github worked fine but the ones built locally (MagiskOnWSALocal) does not as what happens is it will show the WSA Startting up window will show the loading bar and then atter a long while disappears.  The same thing happens when I launch Windows Subsystem for Android Settings and then click on File, I tried what's mentioned by Soetsu09 with both the Intel and the Nvidia GPU on the notebook and then clicked on Files which brings up the Starting WSA window which later exits after some time.  Windows itself does not even see the Amazon app store as Windows Subsystem for Android Settings, Compatibility, Installed apps shows 0 apps found.   I also tried installing WSA from Microsoft Store and it starts up correctly before I uninstalled it and attempted to build it again.  This was built using "./build.sh"

Update:
When I use "./run.sh" instead of "./build.sh", it's working correctly, launches Google Play Store and I have the following apps installed by default:
![image](https://user-images.githubusercontent.com/78993393/224492849-4df4c18b-8836-4e04-97b4-4ecb1f514abd.png)

I didn't see Magisk startup by default

**Reply by lazytiger** [reply]
I tried run.sh, and still the installed app is empty. What should I do?

**Reply by Almighty188** [reply]
What options did you choose in run.sh?

**Reply by lazytiger** [reply]
all default except no amazon app store

**Reply by lazytiger** [reply]
I also tried all default options, there is no luck, still no app installed.

**Reply by Almighty188** [reply]
Did you delete the download folder and also do "git pull" first?

**Reply by lazytiger** [reply]
I uninstalled the WSA, deleted the cloned repository, and did all the things from the beginning again, but still no luck.

**Reply by Almighty188** [reply]
You might want to open a issue.

**Reply by lazytiger** [reply]
I don't think I can supply any helpful information here, I just switched to the official version.

**Reply by Almighty188** [reply]
Just post the logged screen output and it might give some clues.
